src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/core/AbstractCraftingRequestResolver.java

```diff
@@ -145,13 +144,13 @@ public boolean canResolveForBuilding(@NotNull final IRequestManager manager, @No
             return false;
         }
 
-        final boolean isFood = !(request.getRequest() instanceof Food);
+        final boolean isFood = request.getRequest() instanceof Food;
         for (final ICraftingBuildingModule module : building.getModules(ICraftingBuildingModule.class))
         {
             final IRecipeStorage recipe = module.getFirstRecipe(itemStack -> request.getRequest().matches(itemStack));
 
             // If this building is resolving a generic food request, then only allow it to resolve non-smeltables.
-            if (recipe != null && (!isFood || recipe.getIntermediate() != Blocks.FURNACE) && canBuildingCraftStack(building, recipe))
+            if (recipe != null && (!isFood || recipe.getIntermediate() != Blocks.FURNACE) && canBuildingCraftRecipe(building, recipe))
```

Comment:
Is this logic still correct with the non-inverted `isFood`?  Or was it wrong before?

Reply:
Yes